bittide-instances/data/picocom/start.sh

```diff
@@ -8,4 +8,34 @@
 #
 #    /dev/serial/by-id/usb-FTDI_Dual_RS232-if01-port0
 #
-picocom  --baud 921600 --imap lfcrlf --omap lfcrlf $@
+if [[ -z "${PICOCOM_STDOUTLOG}" ]]
+then
+  >&2 echo "envvar \$PICOCOM_STDOUTLOG not set. defaulting to /dev/null."
+  PICOCOM_STDOUTLOG=/dev/null
+else
+  >&2 echo "envvar \$PICOCOM_STDOUTLOG set to '$PICOCOM_STDOUTLOG'."
+  outdir=$(dirname "$PICOCOM_STDOUTLOG")
+  if [[ ! -d "$outdir" ]]
+  then
+    >&2 echo "directory $outdir does not exist. creating it."
+    mkdir -p "$outdir"
+  fi
+fi
+
+if [[ -z "${PICOCOM_STDERRLOG}" ]]
+then
+  >&2 echo "envvar \$PICOCOM_STDERRLOG not set. defaulting to /dev/null."
+  PICOCOM_STDERRLOG=/dev/null
+else
+  >&2 echo "envvar \$PICOCOM_STDERRLOG set to '$PICOCOM_STDERRLOG'."
+  outdir=$(dirname "$PICOCOM_STDERRLOG")
+  if [[ ! -d "$outdir" ]]
+  then
+    >&2 echo "directory $outdir does not exist. creating it."
+    mkdir -p "$outdir"
+  fi
+fi
```

Comment:
I think I'd keep it much simpler, maybe:

```suggestion
set -e

# Default stdout to /dev/null
PICOCOM_STDOUT_LOG="${PICOCOM_STDOUT_LOG:-/dev/null}"
stdout_dir=$(dirname "${PICOCOM_STDOUT_LOG}")
mkdir -p "${stdout_dir}"

# Default stderr to /dev/null
PICOCOM_STDERR_LOG="${PICOCOM_STDERR_LOG:-/dev/null}"
stderr_dir=$(dirname "${PICOCOM_STDERR_LOG}")
mkdir -p "${stderr_dir}"
```

I can see that there is value in warning for undefined variables, but all the other things are way too noisy IMO.

```suggestion
if [[ -z "${PICOCOM_STDOUTLOG}" ]]
then
  >&2 echo "envvar \$PICOCOM_STDOUTLOG not set. defaulting to /dev/null."
  PICOCOM_STDOUTLOG=/dev/null
fi

outdir=$(dirname "$PICOCOM_STDOUTLOG")
mkdir -p "$outdir"

if [[ -z "${PICOCOM_STDERRLOG}" ]]
then
  >&2 echo "envvar \$PICOCOM_STDERRLOG not set. defaulting to /dev/null."
  PICOCOM_STDERRLOG=/dev/null
fi

mkdir -p "$outdir"
```

In any case: 

1. `set -e` prevents silent errors, use it :). 
2. `PICOCOM_STDERRLOG` -> `PICOCOM_STDERR_LOG`
3. I think `$a` -> `${a}` is good practice.